hack/generator/pkg/astbuilder/assignments.go

```diff
@@ -6,26 +6,59 @@
 package astbuilder
 
 import (
+	"fmt"
 	"go/token"
 
 	"github.com/dave/dst"
 )
 
 // SimpleAssignment performs a simple assignment like:
-//     <lhs> := <rhs>       // tok = token.DEFINE
-// or  <lhs> = <rhs>        // tok = token.ASSIGN
-func SimpleAssignment(lhs dst.Expr, tok token.Token, rhs dst.Expr) *dst.AssignStmt {
+//
+//     <lhs> = <rhs>
+//
+func SimpleAssignment(lhs dst.Expr, rhs dst.Expr) *dst.AssignStmt {
 	return &dst.AssignStmt{
 		Lhs: []dst.Expr{
 			dst.Clone(lhs).(dst.Expr),
 		},
-		Tok: tok,
+		Tok: token.ASSIGN,
 		Rhs: []dst.Expr{
 			dst.Clone(rhs).(dst.Expr),
 		},
 	}
 }
 
+// SimpleDeclaration performs a simple assignment like:
+//
+//     <id> := <rhs>
+//
+func SimpleDeclaration(id string, rhs dst.Expr) *dst.AssignStmt {
```

Comment:
We might need to be careful about nomenclature here. [This](https://blog.golang.org/declaration-syntax) and [this](https://gobyexample.com/variables) both use declaration to refer to _just_ the `var x` bit.

In other places in the code we for example have `LocalVariableDeclaration` (which makes a `var <ident> <type>`) and `VariableDeclaration` which makes a `var <ident> <type>` as well (but with a different return type).

It seems like _also_ using `Declaration` to refer to `a := b` might cause confusion?

I think we use the `Definition` word less (and the token is `DEFINE`, not `DECLARE`) - does `SimpleDefinition` do better? Or maybe we need to make it `SimpleDeclareAndAssign` or something?

Reply:
I'm not entirely wedded to the naming ... but ... I used the term declaration because the Go Language feature for `:=` is literally called _Short Variable Declarations_.

Though, I agree that having the token as `DEFINE` is inconsistent.

I'll rename the method to `ShortDeclaration()` to better align with the GoLang terminology.